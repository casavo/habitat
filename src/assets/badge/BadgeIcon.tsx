export const BadgeIcon = () => {
  return (
    <svg
      width="16"
      height="16"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path d="M8.00568 1.33337L14.6667 5.03708V14.6667H1.33337V5.03708L8.00568 1.33337ZM8.00504 3.23921L3.00004 6.01754V13H13V6.01671L8.00504 3.23921Z" />
    </svg>
  );
};
