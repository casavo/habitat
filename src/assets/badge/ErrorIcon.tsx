export const ErrorIcon = () => {
  return (
    <svg
      width="16"
      height="16"
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M12.8 2L14 3.2L9.19976 7.99994L14 12.8L12.8 14L7.99994 9.19976L3.2 14L2 12.8L6.79928 7.99994L2 3.2L3.2 2L7.99994 6.79928L12.8 2Z"
        fill="#BC4646"
      />
    </svg>
  );
};
