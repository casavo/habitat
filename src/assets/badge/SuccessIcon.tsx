export const SuccessIcon = () => {
  return (
    <svg
      width="16"
      height="16"
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M13.3484 2L14.6667 3.1916L6.00701 14L1.33337 9.82676L2.44729 8.41938L5.80333 11.4157L13.3484 2Z"
        fill="#529A46"
      />
    </svg>
  );
};
