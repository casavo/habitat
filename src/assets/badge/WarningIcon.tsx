export const WarningIcon = () => {
  return (
    <svg
      width="16"
      height="16"
      viewBox="0 0 16 16"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.00002 3.99325L13.02 12.6666H2.98002L8.00002 3.99325ZM8.00002 1.33325L0.666687 13.9999H15.3334L8.00002 1.33325ZM8.66669 10.6666H7.33335V11.9999H8.66669V10.6666ZM8.66669 6.66659H7.33335V9.33325H8.66669V6.66659Z"
        fill="#CC9A36"
      />
    </svg>
  );
};
