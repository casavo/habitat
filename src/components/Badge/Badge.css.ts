/* eslint-disable sort-keys-fix/sort-keys-fix */
import { recipe } from "@vanilla-extract/recipes";
import { vars } from "../../utils/theme.css";

export const BadgeContainer = recipe(
  {
    base: {
      alignItems: "center",
      borderRadius: 20,
      display: "inline-flex",
      gap: vars.space[100],
      justifyContent: "center",
      padding: `${vars.space[100]} ${vars.space[200]}`,
    },
    variants: {
      theme: {
        dark: {
          backgroundColor: vars.colors.deepgreen[100],
          color: vars.colors.neutral[0],
          fill: vars.colors.neutral[0],
        },
        light: {
          backgroundColor: vars.colors.offwhite[100],
          color: vars.colors.neutral[100],
          fill: vars.colors.neutral[100],
        },
      },
      status: {
        error: {
          backgroundColor: vars.colors.alertcolors.error[40],
          color: vars.colors.neutral[100],
        },
        success: {
          backgroundColor: vars.colors.alertcolors.success[40],
          color: vars.colors.neutral[100],
        },
        warning: {
          backgroundColor: vars.colors.alertcolors.warning[40],
          color: vars.colors.neutral[100],
        },
      },
    },
  },
  ""
);
