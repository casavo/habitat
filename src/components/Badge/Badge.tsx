import React from "react";
import { ErrorIcon } from "../../assets/badge/ErrorIcon";
import { SuccessIcon } from "../../assets/badge/SuccessIcon";
import { WarningIcon } from "../../assets/badge/WarningIcon";
import { BadgeContainer } from "./Badge.css";
import { Body } from "../Typography";

type BadgeProps = {
  children: React.ReactNode;
  icon?: React.ReactNode;
  status?: "success" | "error" | "warning";
  theme?: "light" | "dark";
};

type StatusIcon = {
  error: React.ReactNode;
  success: React.ReactNode;
  warning: React.ReactNode;
};

const statusIcon: StatusIcon = {
  error: <ErrorIcon />,
  success: <SuccessIcon />,
  warning: <WarningIcon />,
};

export const Badge = ({
  children,
  icon,
  status,
  theme = "light",
}: BadgeProps) => {
  return (
    <span className={BadgeContainer({ status, theme })}>
      {status ? statusIcon[status] : icon}
      <Body size="s">{children}</Body>
    </span>
  );
};
