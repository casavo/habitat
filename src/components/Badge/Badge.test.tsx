import { render } from "@testing-library/react";
import { expect, describe } from "vitest";

import { Badge } from "./Badge.tsx";
import { WarningIcon } from "../../assets/badge/WarningIcon";

describe("Badge component", () => {
  it("renders", () => {
    const { container } = render(<Badge>Click me</Badge>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with a theme", () => {
    const { container } = render(<Badge theme="dark">Click me</Badge>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with a status", () => {
    const { container } = render(<Badge status="warning">Click me</Badge>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with an icon", () => {
    const { container } = render(
      <Badge icon={<WarningIcon />}>Click me</Badge>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
