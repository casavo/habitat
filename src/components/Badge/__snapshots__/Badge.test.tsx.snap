// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge component > renders 1`] = `
<span
  class="Badge__lg8yeu0 Badge_theme_light__lg8yeu2"
>
  <p
    class="Typography__3igy4na Typography_base__3igy4n9 Typography_BodyStyle_noMargin_true__3igy4nb Typography_BodyStyle_size_s__3igy4ne"
  >
    Click me
  </p>
</span>
`;

exports[`Badge component > renders with a status 1`] = `
<span
  class="Badge__lg8yeu0 Badge_status_warning__lg8yeu5 Badge_theme_light__lg8yeu2"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.00002 3.99325L13.02 12.6666H2.98002L8.00002 3.99325ZM8.00002 1.33325L0.666687 13.9999H15.3334L8.00002 1.33325ZM8.66669 10.6666H7.33335V11.9999H8.66669V10.6666ZM8.66669 6.66659H7.33335V9.33325H8.66669V6.66659Z"
      fill="#CC9A36"
    />
  </svg>
  <p
    class="Typography__3igy4na Typography_base__3igy4n9 Typography_BodyStyle_noMargin_true__3igy4nb Typography_BodyStyle_size_s__3igy4ne"
  >
    Click me
  </p>
</span>
`;

exports[`Badge component > renders with a theme 1`] = `
<span
  class="Badge__lg8yeu0 Badge_theme_dark__lg8yeu1"
>
  <p
    class="Typography__3igy4na Typography_base__3igy4n9 Typography_BodyStyle_noMargin_true__3igy4nb Typography_BodyStyle_size_s__3igy4ne"
  >
    Click me
  </p>
</span>
`;

exports[`Badge component > renders with an icon 1`] = `
<span
  class="Badge__lg8yeu0 Badge_theme_light__lg8yeu2"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.00002 3.99325L13.02 12.6666H2.98002L8.00002 3.99325ZM8.00002 1.33325L0.666687 13.9999H15.3334L8.00002 1.33325ZM8.66669 10.6666H7.33335V11.9999H8.66669V10.6666ZM8.66669 6.66659H7.33335V9.33325H8.66669V6.66659Z"
      fill="#CC9A36"
    />
  </svg>
  <p
    class="Typography__3igy4na Typography_base__3igy4n9 Typography_BodyStyle_noMargin_true__3igy4nb Typography_BodyStyle_size_s__3igy4ne"
  >
    Click me
  </p>
</span>
`;
